src/devices/vdml_motors.cpp

```diff
@@ -17,265 +17,662 @@ namespace pros {
-	return motor_set_gearing(_port, gearset);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_set_gearing(_ports[i], gearset);
+	}
+	return motor_set_gearing(_ports[0], gearset);
 }
 
-std::int32_t Motor::set_gearing(const pros::v5::Motor_Gear gearset) const {
+std::int32_t Motor_Group::set_gearing(const pros::v5::Motor_Gear gearset) const {
+	empty_motor_group_check(PROS_ERR);
 	_gearset = gearset;
-	return motor_set_gearing(_port, (motor_gearset_e_t)gearset);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_set_gearing(_ports[i], (motor_gearset_e_t)gearset);
+	}
+	return motor_set_gearing(_ports[0], (motor_gearset_e_t)gearset);
 }
 
-std::int32_t Motor::set_gearing(const pros::Color gearset_color) const {
+std::int32_t Motor_Group::set_gearing(const pros::Color gearset_color) const {
+	empty_motor_group_check(PROS_ERR);
 	_gearset = static_cast<pros::v5::Motor_Gear>(gearset_color);
-	return motor_set_gearing(_port, (motor_gearset_e_t)gearset_color);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_set_gearing(_ports[i], (motor_gearset_e_t)_gearset);
+	}
+	return motor_set_gearing(_ports[0], (motor_gearset_e_t)_gearset);
 }
 
-std::int32_t Motor::set_zero_position(const double position) const {
+std::int32_t Motor_Group::set_zero_position(const double position) const {
+	empty_motor_group_check(PROS_ERR);
 	push_motor_configuration();
-	return motor_set_zero_position(_port, position);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_set_zero_position(_ports[i], position);
+	}
+	return motor_set_zero_position(_ports[0], position);
 }
 
-std::int32_t Motor::set_reversed(const bool reverse) const {
+std::int32_t Motor_Group::set_reversed(const bool reverse) const {
+	empty_motor_group_check(PROS_ERR);
 	_reverse = reverse;
-	return motor_set_reversed(_port, reverse);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_set_reversed(_ports[i], reverse ^ _ports[i] < 0);
+	}
+
+	return motor_set_reversed(_ports[0], reverse ^ _ports[0] < 0);
 }
 
-std::int32_t Motor::set_voltage_limit(const std::int32_t limit) const {
+std::int32_t Motor_Group::set_voltage_limit(const std::int32_t limit) const {
+	empty_motor_group_check(PROS_ERR);
 	push_motor_configuration();
-	return motor_set_voltage_limit(_port, limit);
+	for (int i = 1; i < _ports.size(); i++) {
+		motor_set_voltage_limit(_ports[i], limit);
+	}
+	return motor_set_voltage_limit(_ports[0], limit);
+}
+
+std::vector<std::int8_t> Motor_Group::get_ports() {
+	return _ports;
+}
+
+std::int8_t Motor_Group::size() const {
+	return _ports.size();
+}
+
+void Motor_Group::operator+=(Motor_Group& other) {
+	for (int i = 0; i < other._ports.size(); i++) {
+		_ports.emplace_back(other._ports[i]);
+	}
+}
+
+void Motor_Group::append(Motor_Group& other) {
+	(*this) += other;
+}
+
+void Motor_Group::erase_port(std::int8_t port) {
+	int i = 0;
+	while (i < _ports.size()) {
+		if (std::abs(_ports[i]) == port) {
+			_ports.erase(_ports.begin() + i);
+		} else {
+			i++;
+		}
+	}
```

Comment:
use `std::erase` + `std::remove_if` instead.

Reply:
This probably would be a better solution here.